draft-barnes-acme.md

```diff
@@ -1459,14 +1450,131 @@ TODO
+
+MAC-based recovery only needs to assure these properties in the face of an ACME man in the middle.  MAC-based recovery only uses the ACME channel, and that channel is protected from network attackers by the use of HTTPS.
+
+In provisioning for MAC-based recovery, the recovery key is only accessible to the account private key holder, because it depends on the private ECDH key generated by that entity.  Even though an ACME MitM can observe the key agreement messages in the "recoveryKey" field of the registration transaction, the use of ECDH prevents him from learning the recovery key.
+
+Likewise, the use of HMAC in the MAC-based recovery process proves to the server that the requestor has the recovery key without exposing the recovery key.  The fact that the MAC covers the new account key means the ACME MitM can't replay it to authorize a key of his choosing.
+
+Contact-based recovery uses both the ACME channel and the contact channel.  The provisioning process is only visible to an ACME MitM, and even then, the MitM can only observe the contact information provided.  If the ACME attacker does not also have access to the contact channel, there is no risk.
+
+The security of the contact-based recovery process is entirely dependent on the security of the contact channel.  The details of this will depend on the specific out-of-band technique used by the server.  For example:
+
+* If the server requires a user to click a link in a message sent to a contact address, then the contact channel will need to provide confidentiality and authenticity (of the destination) in order to be secure.  Otherwise, a passive attacker could see the link and click it first, or an active attacker could redirect the message.
+* If the server requires a user to respond to a message sent to a contact address, then the contact channel will need to provide source and destination authenticity.  Otherwise an active attacker can spoof a message from the contact address.
+
+Operators of servers should consider these risks in designing contact-based recovery techniques.  Servers should choose recovery techniques that are robust in the face of interference with the contact channel, and should agree to do recovery only for contacts that provide a minimum level of security.
+
+## Denial-of-Service Considerations
+
+As a protocol run over HTTPS, standard considerations for TCP-based and HTTP-based DoS mitigation also apply to ACME.
+
+At the application layer, ACME requires the server to perform a few potentially expensive operations:
+
+* Every POST request requires public-key signature verification
+* New-registration requests that ask for a recovery key require an ECDH computation
+* Identifier validation transactions require outbound connections to potentially attacker-controlled servers
```

Comment:
You're saying that the ACME validation query can be used as an amplification attack?  "Amplification" is probably the wrong word, since the attacker is going to have to send far more data to the server than the server will send to the validation server.  But this is probably still worth noting.